### PR TITLE
feat: Integrate leave channel resolver in client

### PR DESCRIPTION
**Pull request by woosanggyu**
## Specify project
Client and Server

## Description

To leave the chat room, Click the channel for a long time in the chat list and select 'YES'

## Related Issues

N/A

## Tests

https://user-images.githubusercontent.com/45311830/133582302-6e4742e4-bd8c-44cf-9d5d-b53cb8b51a74.mov

## Checklist

Before you create this PR confirms that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide](https://github.com/dooboolab/hackatalk/blob/master/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] Run `yarn lint && yarn tsc`
- [x] Run `yarn test` or `yarn test -u` if you need to update snapshot.
- [x] I am willing to follow-up on review comments in a timely manner.

[wehack2021] - [cloud]

### CODE REVIEW

**Comment by woosanggyu** [comment]
This "PR" was followed by https://github.com/dooboolab/hackatalk/pull/483

**Comment by woosanggyu** [comment]
This "PR" was followed by [https://github.com/dooboolab/hackatalk/pull/483](url)

File: client/src/components/navigations/MainStackNavigator/index.tsx [hunk 1]
```diff
@@ -96,7 +100,6 @@ const onMessageSubscription = graphql`
         }
         memberships(excludeMe: false) {
           user {
-            id
```
Comment (hyochan) [inline client/src/components/navigations/MainStackNavigator/index.tsx:99]: I think it is better to have the `id` here for future maintenance. The uniqueness is very useful in organizing caches.

File: client/src/components/navigations/MainStackNavigator/index.tsx [hunk 2]
```diff
@@ -32,6 +34,8 @@ import {onMessageUpdater} from '../../../relay/updaters';
 import {requestPermissionsAsync} from 'expo-ads-admob';
 import {useDeviceContext} from '../../../providers';
 
+// import {leave} './LeaveChannelModal';
```
Comment (hyochan) [inline client/src/components/navigations/MainStackNavigator/index.tsx:37]: Please remove the comment on `PR`.

**Review by hyochan — CHANGES_REQUESTED** [review]
When I remove the chat and start chatting again, the

File: client/src/components/navigations/MainStackNavigator/index.tsx (hunk 1 above)
Comment (woosanggyu) [reply client/src/components/navigations/MainStackNavigator/index.tsx:99]: Thank you for your advice.

I will add this opinion on the code.

File: client/src/components/navigations/MainStackNavigator/index.tsx (hunk 2 above)
Comment (woosanggyu) [reply client/src/components/navigations/MainStackNavigator/index.tsx:37]: remove this comment..

**Review by hyochan — APPROVED** [review]
Thanks for the great work!

I might have to change the design a bit tough

**Pull request merged**